app/controllers/task_contributions_controller.rb

```diff
@@ -0,0 +1,94 @@
+# frozen_string_literal: true
+
+class TaskContributionsController < ApplicationController
+  include TaskParameters
+
+  before_action :load_and_authorize_task, except: %i[show edit]
+  before_action :load_and_authorize_task_contribution, except: %i[create new]
+
+  def approve_changes
+    if @task.apply_contribution(@task_contribution)
+      TaskContributionMailer.approval_info(@task_contribution).deliver_later
+      redirect_to @task, notice: t('.success')
+    else
+      redirect_to [@task, @task_contribution], alert: t('.error')
+    end
+  end
+
+  def discard_changes
+    if @task_contribution.close
+      TaskContributionMailer.rejection_info(@task_contribution).deliver_later
+      self_closed = @task_contribution.user == current_user
+      redirect_to [@task, (self_closed ? @task_contribution : nil)], notice: t('.success')
```

Comment:
The flash message for rejecting changes as the original task author could be improved: Currently, the message reads that the decoupling was successful. However, this framing is the one the contributor would expect, not the original task author (for them, it is more about successfully rejecting a change).